src/test/lib/DataTest.cpp

```diff
@@ -2960,6 +2960,116 @@ QuicTestStreamAbortConnFlowControl(
+            Stream->Shutdown(QUIC_STREAM_SHUTDOWN_FLAG_ABORT_SEND);
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+
+    static QUIC_STATUS ConnCallback(_In_ MsQuicConnection*, _In_opt_ void* Context, _Inout_ QUIC_CONNECTION_EVENT* Event) {
+        if (Event->Type == QUIC_CONNECTION_EVENT_PEER_STREAM_STARTED) {
+            new(std::nothrow) MsQuicStream(Event->PEER_STREAM_STARTED.Stream, CleanUpAutoDelete, ServerStreamCallback, Context);
+        }
+        return QUIC_STATUS_SUCCESS;
+    }
+};
+
+void
+QuicTestStreamReliableReset(
+    )
+{
+    /*
+
+    Client initiates stream, connects to Server.
+    Client sets ReliableOffset (send offset) to be 6.
+    Client sends data to server
+    Client immediately calls shutdown.
+    Client should finish sending data up to 6 before actually aborting.
+    Server should get data up to 6 before ACKing the shutdown.
+
+
+    Best way to do this;
+     - artificially delay receiver callback
+     - Peer Event Started
+     - Theoretically you can block receive event...
+
+     Quic Send Flag: QUIC_SEND_FLAG_DELAY_FLAG (not always happens)
+      - Here's a send, you don't have to send it yet.
+      - wait for handshake, sleep(50)
+      - start, send (with delay), (set reliable offset), shutdown (do everything after handshake completes)
+
+    */
+    MsQuicRegistration Registration(true);
+    TEST_TRUE(Registration.IsValid());
+
+    MsQuicSettings ServerSettings;
+    MsQuicSettings ClientSettings;
+    ServerSettings.SetReliableResetEnabled(true);
+    ClientSettings.SetReliableResetEnabled(true);
+    ServerSettings.SetPeerBidiStreamCount(1);
+
+    MsQuicConfiguration ServerConfiguration(Registration, "MsQuicTest", ServerSettings, ServerSelfSignedCredConfig);
+    TEST_QUIC_SUCCEEDED(ServerConfiguration.GetInitStatus());
+
+    MsQuicConfiguration ClientConfiguration(Registration, "MsQuicTest", ClientSettings, MsQuicCredentialConfig());
+    TEST_QUIC_SUCCEEDED(ClientConfiguration.GetInitStatus());
+
+    StreamReliableReset Context;
+    #define BUFFER_SIZE 10000
+    #define RELIABLE_SIZE 1000
+    uint8_t SendDataBuffer[BUFFER_SIZE];
+    uint8_t ReceiveDataBuffer[BUFFER_SIZE];
+
+    QUIC_BUFFER SendBuffer { sizeof(SendDataBuffer), SendDataBuffer };
+    QUIC_BUFFER ReceiveBuffer = { sizeof(ReceiveDataBuffer), ReceiveDataBuffer };
+    Context.ReceivedBufferSize = 0;
+
+    MsQuicAutoAcceptListener Listener(Registration, ServerConfiguration, StreamReliableReset::ConnCallback, &Context);
+    TEST_QUIC_SUCCEEDED(Listener.GetInitStatus());
+    TEST_QUIC_SUCCEEDED(Listener.Start("MsQuicTest"));
+    QuicAddr ServerLocalAddr;
+    TEST_QUIC_SUCCEEDED(Listener.GetLocalAddr(ServerLocalAddr));
+
+    MsQuicConnection Connection(Registration);
+    TEST_QUIC_SUCCEEDED(Connection.GetInitStatus());
+
+    MsQuicStream Stream(Connection, QUIC_STREAM_OPEN_FLAG_NONE, CleanUpManual, StreamReliableReset::ClientStreamCallback, &Context);
+    TEST_QUIC_SUCCEEDED(Stream.GetInitStatus());
```

Comment:
No need to open here. Just move down to where you use the stream, so we can keep all the stream code together.

Reply:
Sounds good. Moved code around.